### ISSUE TITLE
Enable the CompilerSupplier to auto-pull Docker images

### ISSUE THREAD

**Issue by gnidan** [trufflesuite/truffle#1432]
## Issue

Truffle v5's compiler auto-downloading works for solcjs but not for Docker. It's likely that we want to avoid requiring the user to grab the image themselves.

Effectively, when the user does not have the Docker image, Truffle should run `docker pull ethereum/solc:<image-version>`.

Truffle should select the best-match `image-version` for the user's specified config value, similar to how the code works for solcjs. (_see implementation: [`CompilerSupplier.prototype.getCached(version)`](https://github.com/trufflesuite/truffle/blob/next/packages/truffle-compile/compilerSupplier.js#L128-L155)_)

We probably **do not** want to run `docker pull` in the situation where we already have the image, since that adds seemingly-unnecessary overhead.

It is possible that there are reasons why we don't want to do this, either for user-experience or because there are technical limitations that make it harder to implement. I remember @cgewecke mentioning this when he was writing CompilerSupplier initially, but I don't remember his reasoning. Any implementation of this capability should probably keep this in mind and consider UX first.

## Actual Results

```
$ truffle compile
Error: Please pull 0.4.22 from docker before trying to compile with it.
 at CompilerSupplier.errors (compilerSupplier.js:518:10)
 at CompilerSupplier.validateDocker (compilerSupplier.js:362:16)
```

## Environment

* Truffle version (`truffle version`): `v5.0.0-next.15`-ish

**Comment by eggplantzzz** [comment]
Not stale

**Comment by eggplantzzz** [comment]
This was fixed in PR #1651.
